scripts/quests/crystalWar/The_Dawn_of_Delectability.lua

```diff
@@ -0,0 +1,200 @@
+                    player:confirmTrade()
+                    quest:setVar(player, 'Prog', 3)
+                    quest:setVar(player, "DelectabilityDay", vanaDay())
+                end,
+            },
+        },
+    },
+
+    -- Speak with Ranpi before day is up
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3 and
+                quest:getVar(player, "DelectabilityDay") >= vanaDay()
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(982)
+        },
+    },
+
+    -- Speak with Ranpi after a game day wait
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3 and
+                quest:getVar(player, "DelectabilityDay") < vanaDay()
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(980),
+
+            onEventFinish =
+            {
+                [980] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.RANPIMONPI_SPECIALTY)
+                    quest:messageSpecial(watersID.text.KEYITEM_OBTAINED, xi.ki.RANPIMONPI_SPECIALTY)
+                    quest:setVar(player, 'Prog', 4)
+                    quest:setVar(player, "DelectabilityDay", 0)
+                end,
+            },
+        },
+    },
+
+    -- Return to Windurst Waters (S) and feed Ranpi.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 4
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(118),
+
+            onEventFinish =
+            {
+                [118] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.RANPIMONPI_SPECIALTY)
+                    npcUtil.giveKeyItem(player, xi.ki.CULINARY_KNIFE)
+                    quest:messageSpecial(watersID.text.KEYITEM_OBTAINED, xi.ki.CULINARY_KNIFE)
+                    quest:setVar(player, 'Prog', 5)
+                end,
+            },
+        },
+    },
+
+    -- New default text for Ranpi
+    {
+        check = function(player, status, vars)
+            return vars.Prog == 5 or status == QUEST_COMPLETED
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Ranpi-Monpi'] = quest:event(120):replaceDefault(),
+        },
+    },
+
+    -- Return to Windurst Waters and finish the quest
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 5
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(981),
+
+            onEventFinish =
+            {
+                [981] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.CULINARY_KNIFE)
+                    quest:complete(player)
```

Comment:
There should probably be some safety here.  We don't want to delete their keyitem unless the quest successfully completes (it won't if the player's inventory is full)
```
if quest:complete(player) then
    player:delKeyItem(xi.ki.CULINARY_KNIFE)
end
```